temporal-sdk/src/main/java/io/temporal/worker/WorkflowImplementationOptions.java

```diff
@@ -64,36 +68,51 @@ public Builder setFailWorkflowExceptionTypes(
       return this;
     }
 
+    public Builder setActivityOptions(Map<String, ActivityOptions> activityOptions) {
+      this.activityOptions = activityOptions;
+      return this;
+    }
+
     public WorkflowImplementationOptions build() {
       return new WorkflowImplementationOptions(
-          failWorkflowExceptionTypes == null ? new Class[0] : failWorkflowExceptionTypes);
+          failWorkflowExceptionTypes == null ? new Class[0] : failWorkflowExceptionTypes,
+          activityOptions == null ? new HashMap<>() : activityOptions);
     }
   }
 
   private final Class<? extends Throwable>[] failWorkflowExceptionTypes;
+  private final Map<String, ActivityOptions> activityOptions;;
 
-  public WorkflowImplementationOptions(Class<? extends Throwable>[] failWorkflowExceptionTypes) {
+  public WorkflowImplementationOptions(
+      Class<? extends Throwable>[] failWorkflowExceptionTypes,
+      Map<String, ActivityOptions> activityOptions) {
     this.failWorkflowExceptionTypes = failWorkflowExceptionTypes;
+    this.activityOptions = activityOptions;
   }
 
   public Class<? extends Throwable>[] getFailWorkflowExceptionTypes() {
     return failWorkflowExceptionTypes;
   }
 
+  public Map<String, ActivityOptions> getActivityOptions() {
+    return activityOptions;
+  }
+
   @Override
   public String toString() {
-    return "WorkflowImplementationOptions{"
-        + "failWorkflowExceptionTypes="
-        + Arrays.toString(failWorkflowExceptionTypes)
-        + '}';
+    String activityOpsStr = (activityOptions == null) ? "{}" : activityOptions.toString();
+    return String.format(
+        "WorkflowImplementationOptions{failWorkflowExceptionTypes=%s,activityOptions=%s}",
+        Arrays.toString(failWorkflowExceptionTypes), activityOpsStr);
```

Comment:
I usually use IntelliJ to generate toString, equals, hashCode. Press Command-N and select the type of the method to generate.